core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksStore.java

```diff
@@ -0,0 +1,208 @@
+ * initializing the database and performing database backup/restore.
+ */
+public final class RocksStore implements Closeable {
+  private static final Logger LOG = LoggerFactory.getLogger(RocksStore.class);
+
+  private final String mDbPath;
+  private final String mDbBackupPath;
+  private final Collection<ColumnFamilyDescriptor> mColumnFamilyDescriptors;
+  private final DBOptions mDbOpts;
+
+  private RocksDB mDb;
+  /** Map from column name to column handle. */
+  private Map<String, ColumnFamilyHandle> mColumnFamilyHandles;
+
+  /**
+   * @param dbPath a path for the rocks database
+   * @param backupPath a path for taking database backups
+   * @param columnFamilyDescriptors columns to create within the rocks database
+   * @param dbOpts db options
+   */
+  public RocksStore(String dbPath, String backupPath,
+      Collection<ColumnFamilyDescriptor> columnFamilyDescriptors, DBOptions dbOpts) {
+    mDbPath = dbPath;
+    mDbBackupPath = backupPath;
+    mColumnFamilyDescriptors = columnFamilyDescriptors;
+    mDbOpts = dbOpts;
+    new File(mDbBackupPath).mkdirs();
+    try {
+      resetDb();
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * @return the underlying rocksdb instance. The instance changes when clear() is called, so if the
+   *         caller caches the returned db, they must reset it after calling clear()
+   */
+  public synchronized RocksDB getDb() {
+    return mDb;
+  }
+
+  /**
+   * @param columnName a column name
+   * @return the column family handle for the given name. The columns change when clear() is called,
+   *         so if the caller caches the returned column, they must reset it after calling clear()
+   */
+  public synchronized ColumnFamilyHandle getColumn(String columnName) {
+    return mColumnFamilyHandles.get(columnName);
+  }
+
+  /**
+   * Clears and re-initializes the database.
+   */
+  public synchronized void clear() {
+    try {
+      resetDb();
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private synchronized void resetDb() throws RocksDBException {
+    stopDb();
+    formatDbDirs();
+    createDb();
+  }
+
+  private synchronized void stopDb() {
+    if (mDb != null) {
+      try {
+        // Column handles must be closed before closing the db, or an exception gets thrown.
+        mColumnFamilyHandles.values().forEach(handle -> handle.close());
+        mDb.close();
+      } catch (Throwable t) {
+        LOG.error("Failed to close rocks database", t);
+      }
+      mColumnFamilyHandles = null;
+      mDb = null;
+    }
+  }
+
+  private void formatDbDirs() {
+    try {
+      if (FileUtils.exists(mDbPath)) {
+        FileUtils.deletePathRecursively(mDbPath);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    new File(mDbPath).mkdirs();
+  }
+
+  private void createDb() throws RocksDBException {
```

Comment:
What's the guideline for which methods are synchronized?

Reply:
public methods, updated this and the javadoc